### PR TITLE
Added random pausing to continuous ingest

### PR DESCRIPTION
**Pull request by mikewalch**

### CODE REVIEW

File: core/src/main/java/org/apache/accumulo/testing/core/continuous/ContinuousIngest.java [hunk 1]
```diff
@@ -45,11 +45,43 @@
...
+  private static long lastPause;
+  private static long pauseWaitMs;
 
   private static ColumnVisibility getVisibility(Random rand) {
     return visibilities.get(rand.nextInt(visibilities.size()));
   }
 
+  private static boolean pauseEnabled(Properties props) {
+    String value = props.getProperty(TestProps.CI_INGEST_PAUSE_ENABLED);
+    return value != null && value.equals("true");
```
Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/testing/core/continuous/ContinuousIngest.java]: Could use `return Boolean.parseBoolean(value);` here.

File: core/src/main/java/org/apache/accumulo/testing/core/continuous/ContinuousIngest.java [hunk 2]
```diff
@@ -45,11 +45,43 @@
...
+  }
+
+  private static int getPauseWaitMs(Properties props, Random rand) {
+    int waitMin = Integer.parseInt(props.getProperty(TestProps.CI_INGEST_PAUSE_WAIT_MIN));
+    int waitMax = Integer.parseInt(props.getProperty(TestProps.CI_INGEST_PAUSE_WAIT_MAX));
```
Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/testing/core/continuous/ContinuousIngest.java]: Should probably check args to make sure `waitMax >= waitMin` and that both are `>=0`.

File: core/src/main/java/org/apache/accumulo/testing/core/continuous/ContinuousIngest.java [hunk 3]
```diff
@@ -45,11 +45,43 @@
...
+    return (rand.nextInt(waitMax - waitMin) + waitMin) * 1000;
+  }
+
+  private static int getPauseDurationMs(Properties props, Random rand) {
+    int durationMin = Integer.parseInt(props.getProperty(TestProps.CI_INGEST_PAUSE_DURATION_MIN));
+    int durationMax = Integer.parseInt(props.getProperty(TestProps.CI_INGEST_PAUSE_DURATION_MAX));
```
Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/testing/core/continuous/ContinuousIngest.java]: Should check that `max>=min` and both `>=0`

File: core/src/main/java/org/apache/accumulo/testing/core/continuous/ContinuousIngest.java [hunk 4]
```diff
@@ -45,11 +45,43 @@
...
+    return (rand.nextInt(durationMax - durationMin) + durationMin) * 1000;
```
Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/testing/core/continuous/ContinuousIngest.java]: `rand.nextInt(0)` throws an exception. Should check that max is strictly greater than min before using the difference in the `rand.nextInt()`.

File: core/src/main/java/org/apache/accumulo/testing/core/continuous/ContinuousIngest.java [hunk 5]
```diff
@@ -45,11 +45,43 @@
...
+  }
+
+  private static void pauseCheck(Properties props, Random rand) throws InterruptedException {
+    if (pauseEnabled(props)) {
+      if (System.currentTimeMillis() > (lastPause + pauseWaitMs)) {
+        long pauseDurationMs = getPauseDurationMs(props, rand);
+        System.out.println("Pausing ingest for " + pauseDurationMs + " ms - " + System.currentTimeMillis());
+        Thread.sleep(pauseDurationMs);
+        lastPause = System.currentTimeMillis();
```
Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/testing/core/continuous/ContinuousIngest.java]: Should use nanoTime for time durations.

File: core/src/main/java/org/apache/accumulo/testing/core/continuous/ContinuousIngest.java [hunk 6]
```diff
@@ -45,11 +45,43 @@
...
+        pauseWaitMs = getPauseWaitMs(props, rand);
+        System.out.println("Ingest restarted. Will pause again in " + pauseWaitMs + " ms at " + (lastPause + pauseWaitMs));
```
Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/testing/core/continuous/ContinuousIngest.java]: Is STDOUT consistent with the rest of CI output, or should we use a logger?

Comment (mikewalch) [reply core/src/main/java/org/apache/accumulo/testing/core/continuous/ContinuousIngest.java]: STDOUT was consistent but I think a logger is better so I switched everything.

File: core/src/main/java/org/apache/accumulo/testing/core/continuous/ContinuousIngest.java (hunk 5 above)
Comment (mikewalch) [reply core/src/main/java/org/apache/accumulo/testing/core/continuous/ContinuousIngest.java]: Using nanoTime now.

File: core/src/main/java/org/apache/accumulo/testing/core/continuous/ContinuousIngest.java (hunk 2 above)
Comment (mikewalch) [reply core/src/main/java/org/apache/accumulo/testing/core/continuous/ContinuousIngest.java]: Fixed in 84fcd3a630

File: core/src/main/java/org/apache/accumulo/testing/core/continuous/ContinuousIngest.java (hunk 3 above)
Comment (mikewalch) [reply core/src/main/java/org/apache/accumulo/testing/core/continuous/ContinuousIngest.java]: Fixed in 84fcd3a630

File: core/src/main/java/org/apache/accumulo/testing/core/continuous/ContinuousIngest.java (hunk 1 above)
Comment (mikewalch) [reply core/src/main/java/org/apache/accumulo/testing/core/continuous/ContinuousIngest.java]: Fixed in 84fcd3a630

File: core/src/main/java/org/apache/accumulo/testing/core/continuous/ContinuousIngest.java (hunk 4 above)
Comment (mikewalch) [reply core/src/main/java/org/apache/accumulo/testing/core/continuous/ContinuousIngest.java]: I now return min if they are equal.

**Pull request merged**